setup/includes/modinstallsettings.class.php

```diff
@@ -11,31 +11,88 @@
 /**
  * @package setup
  */
-class modInstallSettings {
+class modInstallSettings
+{
     public $install = null;
     public $config = [];
     public $fileName = '';
     public $settings = [];
 
-    function __construct(modInstall &$install,array $config = []) {
+    public function __construct(modInstall &$install, array $config = [])
+    {
         $this->install =& $install;
-        $this->config = array_merge([],$config);
+        $this->config = array_merge([], $config);
 
-        $this->fileName = $this->getCachePath().'settings.cache.php';
+        $this->fileName = $this->getCachePath() . 'settings.cache.php';
         $this->load();
     }
 
     /**
      * @return string
      */
-    protected function getCachePath() {
+    protected function getCachePath()
+    {
         return MODX_CORE_PATH . 'cache/' . (MODX_CONFIG_KEY == 'config' ? '' : MODX_CONFIG_KEY . '/') . 'setup/';
     }
 
     /**
      * @return void
      */
-    protected function rebuildDSN() {
+    public function load()
+    {
+        if (file_exists($this->fileName)) {
+            if (function_exists('opcache_invalidate')) {
+                if (stripos(__FILE__, ini_get('opcache.restrict_api')) !== false) {
```

Comment:
I don't get the clue on this patch. You are searching the path of the current file for the value of opcache.restrict_api php setting?

Reply:
Yes, maybe I should use $_SERVER['SCRIPT_FILENAME'] instead. 
But see https://serverfault.com/a/998876 for more info.

Something similar is done in Joomla as well: https://github.com/joomla/joomla-cms/blob/ebd801ade267cfc367979972c7876480c6343248/administrator/components/com_joomlaupdate/extract.php#L1791

Reply:
Ok. Thanks.

Reply:
@Jako To clear the confusion:

The description for the `opcache.restrict_api` setting:
> Allows calling OPcache API functions only from PHP scripts which path is started from specified string. The default "" means no restriction.

We are allowed to use `opcache_` functions if the path defined in `opcache.restrict_api` can be found in the path of the current file/script. 

> stripos(string $haystack, string $needle, int $offset = 0)
> Find the numeric position of the first occurrence of `needle` in the `haystack` string.